code/game/objects/items/weapons/melee/melee_misc.dm

```diff
@@ -45,6 +45,321 @@
 	AddComponent(/datum/component/parry, _stamina_constant = 2, _stamina_coefficient = 0.5, _parryable_attack_types = NON_PROJECTILE_ATTACKS)
 	RegisterSignal(src, COMSIG_PARENT_QDELETING, PROC_REF(alert_admins_on_destroy))
 
+//Security Sword
+/obj/item/melee/secsword
+	name = "securiblade"
+	desc = "This nanite-honed blade comes with a battery in the hilt, to add an electric shock to every swing."
+	lefthand_file = 'icons/mob/inhands/weapons_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons_righthand.dmi'
+	var/base_icon = "secsword"
+	icon_state = "secsword0"
+	item_state = "secsword0"
+	flags = CONDUCT
+	force = 15
+	var/stam_damage = 35 // 3 hits to stamcrit
+	var/burn_damage = 10
+	throwforce = 5
+	w_class = WEIGHT_CLASS_BULKY
+	sharp = TRUE
+	origin_tech = "combat=4"
+	attack_verb = list("stabbed", "slashed", "sliced")
+	hitsound = 'sound/weapons/bladeslice.ogg'
+	materials = list(MAT_METAL = 1000)
+	needs_permit = TRUE
+	/// How much power does it cost to stun someone
+	var/stam_hitcost = 1000
+	/// How much power does it cost to burn someone
+	var/burn_hitcost = 1500
+	/// the initial cooldown tracks the time between stamina damage. tracks the world.time when the baton is usable again.
+	var/cooldown = 3.5 SECONDS
+	var/obj/item/stock_parts/cell/high/cell = null
+	/// Sword state 0 - off, 1 - stam, 2 - burn
+	var/state = 0
+
+/obj/item/melee/secsword/examine(mob/user)
+	. = ..()
+	if(cell)
+		. += "<span class='notice'>It is [round(cell.percent())]% charged.</span>"
+	. += "<span class='notice'>Can be recharged with a recharger</span>"
+
+/obj/item/melee/secsword/update_icon_state()
+	if(!cell)
+		icon_state = "[base_icon]3"
+		item_state = "[base_icon]0"
+	else
+		icon_state = "[base_icon][state]"
+		item_state = "[base_icon][state]"
+
+/obj/item/melee/secsword/Initialize(mapload)
+	. = ..()
+	link_new_cell()
+	AddComponent(/datum/component/parry, _stamina_constant = 2, _stamina_coefficient = 1, _parryable_attack_types = NON_PROJECTILE_ATTACKS)
+
+/obj/item/melee/secsword/proc/link_new_cell(unlink = FALSE)
+	if(unlink)
+		cell = null
+	else
+		cell = new(src)
+
+/obj/item/melee/secsword/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/stock_parts/cell))
+		var/obj/item/stock_parts/cell/C = I
+		if(cell)
+			to_chat(user, "<span class='warning'>[src] already has a cell!</span>")
+			return
+		if(C.maxcharge < stam_hitcost)
+			to_chat(user, "<span class='warning'>[src] requires a higher capacity cell!</span>")
+			return
+		if(!user.unEquip(I))
+			return
+		I.forceMove(src)
+		cell = I
```

Comment:
this cell needs to be properly GC'd when QDEL'd. If it gets deleted, the sword will still hold a reference to it and there will be a GC failure.